modules/core/src/passes/layers-pass.ts

```diff
@@ -349,27 +356,34 @@ export default class LayersPass extends Pass {
   ): any {
     // @ts-expect-error TODO - assuming WebGL context
     const devicePixelRatio = this.device.canvasContext.cssToDeviceRatio();
-
-    const moduleParameters = Object.assign(
-      Object.create(layer.internalState?.propsInTransition || layer.props),
-      {
-        autoWrapLongitude: layer.wrapLongitude,
+    const layerProps = layer.internalState?.propsInTransition || layer.props;
+
+    const moduleParameters = {
+      layer: layerProps,
+      picking: {
+        isActive: false
+      } satisfies PickingProps,
+      project: {
         viewport: layer.context.viewport,
-        mousePosition: layer.context.mousePosition,
-        picking: {
-          isActive: 0
-        },
-        devicePixelRatio
-      }
-    );
+        devicePixelRatio,
+        modelMatrix: layerProps.modelMatrix,
+        coordinateSystem: layerProps.coordinateSystem,
+        coordinateOrigin: layerProps.coordinateOrigin,
+        autoWrapLongitude: layer.wrapLongitude
+      } satisfies ProjectProps
+    };
```

Comment:
Again an `as const satisifies ...` statement here would solidify the type checks.